File: src/main/java/javalin/core/ErrorMapper.java
// Javalin - https://javalin.io
// Copyright 2017 David Åse
// Licensed under Apache 2.0: https://github.com/tipsy/javalin/blob/master/LICENSE

package javalin.core;

import java.util.HashMap;
import java.util.Map;

import javalin.ErrorHandler;
import javalin.Request;
import javalin.Response;

public class ErrorMapper {

    private Map<Integer, ErrorHandler> errorHandlerMap;

    public ErrorMapper() {
        this.errorHandlerMap = new HashMap<>();
    }

    public void put(Integer statusCode, ErrorHandler handler) {
        this.errorHandlerMap.put(statusCode, handler);
    }

    public void clear() {
        this.errorHandlerMap.clear();
    }

    void handle(int statusCode, Request request, Response response) {
        ErrorHandler errorHandler = errorHandlerMap.get(statusCode);
        if (errorHandler != null) {
            errorHandler.handle(request, response);
        }
    }
}

File: src/main/kotlin/io.javalin/core/ErrorMapper.kt
/*
 * Javalin - https://javalin.io
 * Copyright 2017 David Åse
 * Licensed under Apache 2.0: https://github.com/tipsy/javalin/blob/master/LICENSE
 */

package io.javalin.core

import io.javalin.ErrorHandler
import io.javalin.Request
import io.javalin.Response
import java.util.*

class ErrorMapper {

    private val errorHandlerMap: MutableMap<Int, ErrorHandler>

    init {
        this.errorHandlerMap = HashMap<Int, ErrorHandler>()
    }

    fun put(statusCode: Int, handler: ErrorHandler) {
        this.errorHandlerMap.put(statusCode, handler)
    }

    fun clear() {
        this.errorHandlerMap.clear()
    }

    fun handle(statusCode: Int, request: Request, response: Response) {
        val errorHandler = errorHandlerMap[statusCode]
        errorHandler?.handle(request, response)
    }
}
